adobe/react-spectrum
# Fix restoreTextSelection on pointerup event

Closes #5549

Hello, this PR corrects `restoreTextSelection()` to properly remove `user-select: none` on `pointerup` events. 

I originally [suggested](https://github.com/adobe/react-spectrum/issues/5549#issuecomment-1850139901) to change the condition `target.style.userSelect === 'none' && targetOldUserSelect` to `target.style.userSelect === 'none' && targetOldUserSelect !== undefined` so that the value `''` passes the test, but I realized `targetOldUserSelect !== undefined` is unnecessary since we have an outer if statement that checks `modifiedElementMap.has(target)` so `targetOldUserSelect` is never `undefined`.

https://github.com/adobe/react-spectrum/blob/5ce9a29250e7cd009d73d9d1d36aa1010d7099b9/packages/%40react-aria/interactions/src/textSelection.ts#L83-L98

So instead in this PR, I simply removed `&& targetOldUserSelect` from the condition (and I added a small test)

## ✅ Pull Request Checklist:


## 📝 Test Instructions:
1. Go to http://localhost:9003/?path=/story/breadcrumbs--default&providerSwitcher-express=false&strict=true
2. Open up a browser devtools, and select the last breadcrumb item.
3. Click on the last breadcrumb item and see if `user-select: none;` is properly added/removed during pointerdown and pointerup.

## 🧢 Your Project:
